poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -330,16 +330,28 @@ public void syncPowerIQAssetsMetaData(FacilitySoftwareConfig powerIQ) {
       }
       LocationInfo location = getLocationInfo(client);
       List<Asset> pdusFromFlowgate = restClient.getAllAssetsBySourceAndType(powerIQ.getId(), AssetCategory.PDU);
-      logger.info("Size :" + pdusFromFlowgate.size());
       Map<String, Asset> pduIDAndAssetMap = getPDUIDAndAssetMap(pdusFromFlowgate);
-      logger.info("Map Size :" + pduIDAndAssetMap.size());
-
-      savePduAssetsToFlowgate(pduIDAndAssetMap, powerIQ.getId(), client, location);
+      boolean hasNewPduToSave =
+            savePduAssetsToFlowgate(pduIDAndAssetMap, powerIQ.getId(), client, location);
       logger.info("Finish sync PDU metadata for " + powerIQ.getName());
 
       Map<String, Asset> exsitingSensorAssets = getAssetsFromWormhole(powerIQ.getId());
+      pdusFromFlowgate = restClient.getAllAssetsBySourceAndType(powerIQ.getId(), AssetCategory.PDU);
+      pduIDAndAssetMap = getPDUIDAndAssetMap(pdusFromFlowgate);
       saveSensorAssetsToFlowgate(exsitingSensorAssets, client, powerIQ.getId(), pduIDAndAssetMap, location);
```

Comment:
can we refactor the saveSensorAssetsToFlowgate  let's remove the parameter pduIDAssetMap,  

since we can get the data directly inside the method.

Reply:
Done